Mesh_2/test/Mesh_2/reproductibility.cpp

```diff
@@ -0,0 +1,106 @@
+
+#include <fstream>
+#include <iostream>
+
+using K = CGAL::Exact_predicates_inexact_constructions_kernel;
+using Vb = CGAL::Triangulation_vertex_base_2<K>;
+using Fb = CGAL::Delaunay_mesh_face_base_2<K>;
+using Tds = CGAL::Triangulation_data_structure_2<Vb, Fb>;
+using CDT = CGAL::Constrained_Delaunay_triangulation_2<K, Tds, CGAL::Exact_predicates_tag>;
+using Criteria = CGAL::Delaunay_mesh_size_criteria_2<CDT>;
+using Mesher =  CGAL::Delaunay_mesher_2<CDT, Criteria>;
+
+using Vertex_handle = CDT::Vertex_handle;
+using Point = CDT::Point;
+
+int main(int argc, char* argv[])
+{
+  std::string path = argv[0];
+  path = path.substr(0, path.rfind('/') + 1);
+
+  std::cout << "Current dir:" << path << std::endl;
+
+  auto triangulate = [&path](int index)
+  {
+    CDT cdt;
+
+    auto write_tr = [&](const std::string& filename)
+    {
+//       std::ofstream file(path + filename + "_" + std::to_string(index) + ".off");
+//
+//       cdt.file_output(file);
+//       file.close();
+    };
+
+    Vertex_handle va = cdt.insert(Point(-0.74397572, -0.54545455));
+    Vertex_handle vb = cdt.insert(Point(-0.13526831, -1));
+    Vertex_handle vc = cdt.insert(Point(0.067634156, -1));
+    Vertex_handle vd = cdt.insert(Point(0.33817078, -0.54545455));
+    Vertex_handle ve = cdt.insert(Point(0.74397572, 0.27272727));
+    Vertex_handle vf = cdt.insert(Point(0.74397572, 0.54545455));
+    Vertex_handle vg = cdt.insert(Point(0.067634156, 1));
+    Vertex_handle vh = cdt.insert(Point(-0.13526831, 1));
+    Vertex_handle vi = cdt.insert(Point(-0.74397572, -0.18181818));
+
+    cdt.insert_constraint(va, vb);
+    cdt.insert_constraint(vb, vc);
+    cdt.insert_constraint(vc, vd);
+    cdt.insert_constraint(vd, ve);
+    cdt.insert_constraint(ve, vf);
+    cdt.insert_constraint(vf, vg);
+    cdt.insert_constraint(vg, vh);
+    cdt.insert_constraint(vh, vi);
+    cdt.insert_constraint(vi, va);
+
+    const std::vector<Point> points{
+      Point(0.65605132, 0.43821259),
+      Point(0.23073753, -0.4476739),
+      Point(-0.037496007, -0.93636364),
+      Point(-0.00095596601, 0.88181818),
+      Point(-0.62452925, -0.30720903),
+      Point(-0.69663181, -0.45045525),
+    };
+
+    cdt.insert(points.cbegin(), points.cend());
+
+    std::cout << "Meshing: " << index << std::endl;
+
+    std::cout << "Number of vertices before: " << cdt.number_of_vertices() << std::endl;
+
+    write_tr("before_refine");
+
+    Mesher mesher(cdt);
+    mesher.set_criteria(Criteria(0.125, 0.05*std::sqrt(2)));
+
+//     mesher.clear_seeds();
+//     mesher.init();
+
+    mesher.refine_mesh();
+
+    write_tr("after_refine");
+
+    std::cout << "Number of vertices after: " << cdt.number_of_vertices() << std::endl;
+
+    return cdt.number_of_vertices();
+  };
+
+  const size_t ref_number_of_vertices = triangulate(0);
+
+  for (int i = 1; i < 20; ++i)
+  {
+    const size_t number_of_vertices = triangulate(i);
+    if (ref_number_of_vertices != number_of_vertices)
+      return 1;
+  }
```

Comment:
Instead of comparing just the number of vertices, I think you should compare the triangulations. You can use the I/O, for that:

```c++
  constexpr int number_of_iterations = 20;
  std::vector<std::string> triangulation_strings(number_of_iterations);
  for (int i = 0; i < number_of_iterations; ++i)
  {
    auto cdt = triangulate(i);
    std::stringstream ss;
    ss << cdt;
    triangulation_strings[i] = ss.str();
    if (i > 0 && triangulation_strings[i] != triangulation_strings[0])
      return 1;
  }
```

(I have not tested.) `triangulate(int)` will also have to be changed to return the triangulation.

Reply:
I just integrated the comparison of the triangulations.

Reply:
And your implementation is better than mine. :+1: